File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/auth/login/LoginContract.java
package ua.com.foxminded.locationtrackera.ui.auth.login;

import ua.com.foxminded.locationtrackera.mvi.FragmentContract;

public class LoginContract {

    public interface ViewModel extends FragmentContract.ViewModel<LoginScreenState> {
        void login(String email, String password);
    }

    public interface View extends FragmentContract.View {
        void showProgress();

        void proceedToNextScreen();

        void showFailureToastMessage();

        void showEmailAndPasswordError(int emailError, int passwordError);
    }

    public interface Host extends FragmentContract.Host {
    }
}

File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/auth/login/LoginContract.kt
package ua.com.foxminded.locationtrackera.ui.auth.login

import ua.com.foxminded.locationtrackera.mvi.fragments.FragmentContract
import ua.com.foxminded.locationtrackera.ui.auth.login.state.LoginScreenEffect
import ua.com.foxminded.locationtrackera.ui.auth.login.state.LoginScreenState

class LoginContract {
    interface ViewModel : FragmentContract.ViewModel<LoginScreenState, LoginScreenEffect> {
        fun login(email: String?, password: String?)
    }

    interface View : FragmentContract.View {
        fun setProgressVisibility(isProgressVisible: Boolean)
        fun proceedToNextScreen()
        fun showFailureToastMessage()
        fun showEmailAndPasswordError(emailError: Int, passwordError: Int)
    }

    interface Host : FragmentContract.Host
}
